hasher-matcher-actioner/hmalib/lambdas/api/api_root.py

```diff
@@ -101,19 +144,56 @@ def lambda_handler(event, context):
 
 
 # TODO move to utils library
-def check_db_for_matches():
+def gen_matches() -> t.List:
+    table = dynamodb.Table(DYNAMODB_TABLE)
+    records = PDQMatchRecord.get_from_time_range(table)
+    results = [
+        {
+            "content_id": record.content_id,
+            "signal_id": record.signal_id,
+            "signal_source": record.signal_source,
+            "updated_at": record.updated_at.isoformat(),
+            "reactions": "TODO",
+        }
+        for record in records
+    ]
+    return results
+
+
+def gen_match_details(content_id: str) -> t.List:
+    if content_id is None:
+        return []
     table = dynamodb.Table(DYNAMODB_TABLE)
-    result = table.scan(
-        ProjectionExpression="SK,PK",
-        FilterExpression=Attr("SK").begins_with("te#"),
+    records = PDQMatchRecord.get_from_content_id(
+        table, f"{IMAGE_FOLDER_KEY}{content_id}"
     )
-    return result.get("Items")
+    results = [
+        {
+            "content_id": record.content_id,
+            "content_hash": record.content_hash,
+            "signal_id": record.signal_id,
+            "signal_hash": record.signal_hash,
+            "signal_source": record.signal_source,
+            "updated_at": record.updated_at.isoformat(),
+        }
+        for record in records
+    ]
+    return results
 
 
-def check_db_for_hash(key):
+def gen_hash(content_id: str) -> t.Dict:
+    if content_id is None:
+        return {}
     table = dynamodb.Table(DYNAMODB_TABLE)
-    result = table.scan(
-        ProjectionExpression="ContentHash",
-        FilterExpression=(Attr("PK").eq(f"c#images/{key}") & Attr("SK").eq("type#pdq")),
+    record = PipelinePDQHashRecord.get_from_content_id(
+        table, f"{IMAGE_FOLDER_KEY}{content_id}"
+    )
+    return (
+        {
+            "content_id": record.content_id,
```

Comment:
Nit: Generally avoid inline if statements, they can make the control flow harder to read - in this case my initial comment was asking why this had been wrapped with parenthesis.
```
if not record:
  return {}
return {
  ...
}
```